src/transformers/agents/python_interpreter.py

```diff
@@ -666,14 +765,13 @@ def evaluate_python_code(
         state = {}
     result = None
     state["print_outputs"] = ""
-
-    for idx, node in enumerate(expression.body):
+    for node in expression.body:
         try:
             line_result = evaluate_ast(node, state, tools, authorized_imports)
         except InterpretorError as e:
-            msg = f"You tried to execute the following code:\n{code}\n"
-            msg += f"You got these outputs:\n{state['print_outputs']}\n"
-            msg += f"Evaluation stopped at line '{node}' because of the following error:\n{e}"
+            msg = f"Evaluation stopped at line '{ast.get_source_segment(code, node)}' because of the following error:\n{e}"
```

Comment:
This replaces error messages `"Evaluation stopped at line 'ast.Import'"` with `"Evaluation stopped at line 'import numpy as np'"`, which is much clearer for both the user and the agent LLM.